src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -1068,6 +1062,19 @@ def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonD
             # and it matches the spec.
             port = spec.get_port()
 
+        if spec.generate_cert:
+            cert, key = self.mgr.cert_mgr.ssl_certs.generate_cert(
```

Comment:
the cert_mgr class has already a method called `generate_cert`, the code has to be:

```suggestion
            cert, key = self.mgr.cert_mgr.generate_cert(
```